core/jazz_es-kinesis-log-streamer/components/utils.js

```diff
@@ -0,0 +1,298 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+/**
+    Helper functions for Cloud Logs Streamer
+  @module: utils.js
+  @description: Helper functions for using Regex patterns, etc.
+  @version: 1.0
+**/
+
+const crypto = require('crypto');
+const logger = require("../components/logger.js");
+const config = require("../config/global_config.json")
+var truncate = require('unicode-byte-truncate');
+
+// Helper functions
+
+function getInfo(messages, patternStr) {
+	let pattern = new RegExp(patternStr);
+	let result = "";
+	if(messages){
+		for (let i = 0, len = messages.length; i < len; i++) {
+			let _tmp = pattern.exec(messages[i].message);
+			if(_tmp && _tmp[1]) {
+				logger.debug("found match..:"+_tmp[1]);
+				result = _tmp[1];
+				break;
+			}
+		}
+	}
+	return result;
+}
+
+function getSubInfo(message, patternStr, index) {
+	let pattern = new RegExp(patternStr);
+	let result = "";
+	if(message){
+		let _tmp = pattern.exec(message);
+		if(_tmp && _tmp[index]) {
+			logger.debug("found match..:"+_tmp[index]);
+			result = _tmp[index];
+		}
+	}
+	return result;
+}
+
+function hmac(key, str, encoding) {
+    return crypto.createHmac('sha256', key).update(str, 'utf8').digest(encoding);
+}
+
+function hash(str, encoding) {
+    return crypto.createHash('sha256').update(str, 'utf8').digest(encoding);
+}
+
+function getApiLogsData(payload) {
+	let bulkRequestBody = '', data = {}, indexName = "apilogs";
+	data.timestamp = new Date();
+	data.platform_log_group = payload.logGroup;
+	data.platform_log_stream = payload.logStream;
+	data.environment = getSubInfo(payload.logGroup, config.PATTERNS.environment, 2);
+	data.request_id = getInfo(payload.logEvents, config.PATTERNS.request_id);
+	data.method = getInfo(payload.logEvents, config.PATTERNS.method);
+	if (!data.method) {
+		data.method = "GET"; 
+	} // Cloudwatch do not have method info for get!
+	
+	let apiDomainAndService = getInfo(payload.logEvents, config.PATTERNS.domain_service);
+	let _apiDomain = apiDomainAndService.substring(0, apiDomainAndService.indexOf("/"));
```

Comment:
why underscore for this var but not for others?

Reply:
replaced with apiDomain